src/Panel/User.php

```diff
@@ -73,19 +73,40 @@ public function dropdown(array $options = []): array
 			'disabled' => $this->isDisabledDropdownOption('changeRole', $options, $permissions)
 		];
 
+		$result[] = [
+			'dialog'   => $url . '/changeLanguage',
+			'icon'     => 'translate',
+			'text'     => I18n::translate('user.changeLanguage'),
+			'disabled' => $this->isDisabledDropdownOption('changeLanguage', $options, $permissions)
+		];
+
+		$result[] = '-';
+
 		$result[] = [
 			'dialog'   => $url . '/changePassword',
 			'icon'     => 'key',
 			'text'     => I18n::translate('user.changePassword'),
 			'disabled' => $this->isDisabledDropdownOption('changePassword', $options, $permissions)
 		];
 
-		$result[] = [
-			'dialog'   => $url . '/changeLanguage',
-			'icon'     => 'translate',
-			'text'     => I18n::translate('user.changeLanguage'),
-			'disabled' => $this->isDisabledDropdownOption('changeLanguage', $options, $permissions)
-		];
+		if ($this->model->kirby()->system()->is2FAWithTOTP() === true) {
+			if ($account || $this->model->kirby()->user()->isAdmin()) {
+				if ($this->model->secret('totp') !== null) {
+					$result[] = [
+						'dialog'   => $url . '/totp/disable',
+						'icon'     => 'qr-code',
+						'text'     => I18n::translate('login.totp.disable.option'),
+					];
+				} elseif ($account) {
+					$result[] = [
+						'dialog'   => $url . '/totp/activate',
+						'icon'     => 'qr-code',
+						'text'     => I18n::translate('login.totp.activate.option'),
+						'disabled' => !$account
```

Comment:
Can this line ever take effect with the `elseif ($account)` above? Looks like a leftover.